### ISSUE TITLE
bug: Temp dir name should be normalized

### ISSUE THREAD

**Issue by stevapple** [mkdocstrings/griffe#324]
### Description of the bug
When the branch name to checkout contains special characters (most notably, `/`), griffe will fail to create a temp directory for it.

### To Reproduce
```
$ griffe check module_name -a origin/main
```

### Full traceback

<details><summary>Full traceback</summary>

```python
+ griffe check module -a origin/main
Traceback (most recent call last):
  File "/usr/local/bin/griffe", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/lib/python3.12/site-packages/_griffe/cli.py", line 560, in main
    return commands[subcommand](**opts_dict)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/_griffe/cli.py", line 475, in check
    old_package = load_git(
                  ^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/_griffe/loader.py", line 881, in load_git
    with tmp_worktree(repo, ref) as worktree:
  File "/usr/local/lib/python3.12/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/_griffe/git.py", line 107, in tmp_worktree
    with TemporaryDirectory(prefix=f"{_WORKTREE_PREFIX}{repo_name}-{ref}-") as tmp_dir:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/tempfile.py", line 882, in __init__
    self.name = mkdtemp(suffix, prefix, dir)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/tempfile.py", line 384, in mkdtemp
    _os.mkdir(file, 0o700)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/griffe-worktree-package-origin/main-f19382b7'
```

</details>

### Expected behavior

The tool should be able to check against arbitrary branch.

**Comment by pawamoy** [comment]
Hi @stevapple, thanks for the report! Looks definitely like something that must be fixed. On it.

**Comment by pawamoy** [comment]
Fixed in v1.3.2 :slightly_smiling_face:
